sql/core/src/test/scala/org/apache/spark/sql/execution/arrow/ArrowConvertersSuite.scala

```diff
@@ -792,6 +793,76 @@ class ArrowConvertersSuite extends SharedSQLContext with BeforeAndAfterAll {
     collectAndValidate(df, json, "binaryData.json")
   }
 
+  test("date type conversion") {
+    val json =
+      s"""
+         |{
+         |  "schema" : {
+         |    "fields" : [ {
+         |      "name" : "date",
+         |      "type" : {
+         |        "name" : "date",
+         |        "unit" : "DAY"
+         |      },
+         |      "nullable" : true,
+         |      "children" : [ ],
+         |      "typeLayout" : {
+         |        "vectors" : [ {
+         |          "type" : "VALIDITY",
+         |          "typeBitWidth" : 1
+         |        }, {
+         |          "type" : "DATA",
+         |          "typeBitWidth" : 32
+         |        } ]
+         |      }
+         |    } ]
+         |  },
+         |  "batches" : [ {
+         |    "count" : 4,
+         |    "columns" : [ {
+         |      "name" : "date",
+         |      "count" : 4,
+         |      "VALIDITY" : [ 1, 1, 1, 1 ],
+         |      "DATA" : [ -1, 0, 16533, 16930 ]
+         |    } ]
+         |  } ]
+         |}
+       """.stripMargin
+
+    val sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS z", Locale.US)
+    val d1 = new Date(-1)  // "1969-12-31 13:10:15.000 UTC"
+    val d2 = new Date(0)  // "1970-01-01 13:10:15.000 UTC"
+    val d3 = new Date(sdf.parse("2015-04-08 13:10:15.000 UTC").getTime)
+    val d4 = new Date(sdf.parse("2016-05-09 12:01:01.000 UTC").getTime)
+
+    // Date is created unaware of timezone, but DateTimeUtils force defaultTimeZone()
+    assert(DateTimeUtils.toJavaDate(DateTimeUtils.fromJavaDate(d2)).getTime == d2.getTime)
```

Comment:
@holdenk @cloud-fan I'm trying out the DateType conversion and ran into this problem.  The Dataset encoder uses `DateTimeUtils.toJavaDate` and `fromJavaDate` similar to above, and this forces a `defaultTimeZone()` when working with the data.  So a value `new Date(0)` should be the epoch, but in my timezone it forces it to be the day before and the test above will not pass.  

What are your thoughts on this, should the conversion to Arrow assume the defaultTimeZone()?  is this something that should be fixed first in Spark?  Thanks!